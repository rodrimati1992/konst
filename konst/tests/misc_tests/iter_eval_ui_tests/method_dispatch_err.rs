const _: () = {
    konst::iter::eval!();
};

const _: () = {
    konst::iter::eval!(0usize..10, asdasdasd());
};

const _: () = {
    konst::iter::eval!(0usize..10, enumerate);
};

const _: () = {
    konst::iter::eval!(0usize..10, asdasdasd);
};

const _: () = {
    konst::iter::eval!(0usize..10, enumerate().foo());
};

const _: () = {
    konst::iter::eval!(0usize..10, enumerate(),nth(3),bar());
};


fn main(){}