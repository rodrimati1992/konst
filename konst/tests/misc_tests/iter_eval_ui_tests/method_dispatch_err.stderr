error: expected iterator argument
 --> tests/misc_tests/iter_eval_ui_tests/method_dispatch_err.rs:2:5
  |
2 |     konst::iter::eval!();
  |     ^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: method `asdasdasd` is unsupported, expected one of:
       - map
       - enumerate
       - copied
       - skip
       - skip_while
       - take
       - take_while
       - filter
       - filter_map
       - zip
       - rev
       - flatten
       - flat_map
       - find
       - rfind
       - find_map
       - position
       - all
       - any
       - nth
       - next
       - count
       - for_each
       - fold
       - rfold
       - cmp
       - eq
       - ne
       - ge
       - gt
       - le
       - lt
       - is_sorted
       - is_sorted_by
       - is_sorted_by_key
       - max
       - min
       - max_by
       - min_by
       - max_by_key
       - min_by_key
 --> tests/misc_tests/iter_eval_ui_tests/method_dispatch_err.rs:6:5
  |
6 |     konst::iter::eval!(0usize..10, asdasdasd());
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)

error: unexpected end of macro invocation
  --> tests/misc_tests/iter_eval_ui_tests/method_dispatch_err.rs:10:45
   |
10 |     konst::iter::eval!(0usize..10, enumerate);
   |                                             ^ missing tokens in macro arguments
   |
note: while trying to match `(`
  --> src/iter/iter_eval_macro.rs
   |
   |         $($(,)+ $methods:ident ($($args:tt)*))*
   |                                ^

error: unexpected end of macro invocation
  --> tests/misc_tests/iter_eval_ui_tests/method_dispatch_err.rs:14:45
   |
14 |     konst::iter::eval!(0usize..10, asdasdasd);
   |                                             ^ missing tokens in macro arguments
   |
note: while trying to match `(`
  --> src/iter/iter_eval_macro.rs
   |
   |         $($(,)+ $methods:ident ($($args:tt)*))*
   |                                ^

error: no rules expected `.`
  --> tests/misc_tests/iter_eval_ui_tests/method_dispatch_err.rs:18:47
   |
18 |     konst::iter::eval!(0usize..10, enumerate().foo());
   |                                               ^ no rules expected this token in macro call
   |
   = note: while trying to match sequence start

error: Cannot consume the iterator multiple times, first consumed by `nth` method
  --> tests/misc_tests/iter_eval_ui_tests/method_dispatch_err.rs:22:5
   |
22 |     konst::iter::eval!(0usize..10, enumerate(),nth(3),bar());
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in the macro `$crate::iter::__eval2_lowering` which comes from the expansion of the macro `konst::iter::eval` (in Nightly builds, run with -Z macro-backtrace for more info)
